stats/stats-server/src/read_service.rs

```diff
@@ -82,6 +82,9 @@ impl StatsService for ReadService {
                 value: point.value,
             })
             .collect();
+
+        // remove last data point, because it can be partially updated
+        data.pop();
```

Comment:
Maybe pop only if date is current day? Like add filter data.filter(|v| v.date < now().date)

In that case we are sure that last value is partially updated, and if blockchain didn't produce something for one day, we will return prev value

Reply:
When we update, our invariant is "last value can be partially updated" (that's why we always update it)
Here, you want to specify, that last value is partial only if it's equal to current day, which is untrue (for example, what if some chart updates once in two days? or service was down for a day?)

> if blockchain didn't produce something for one day
you mean if blockscout stop indexing? then, if it was stopped in the middle of the day, we still have partial updated value
if you mean that no i.e. transactions were produced, then we should fill the date with zero, and not skip it